docs/source/en/pipeline_tutorial.mdx

```diff
@@ -33,101 +33,202 @@ While each task has an associated [`pipeline`], it is simpler to use the general
+With that parameter, the model inferred the texts, but also outputted along **when** the various words were pronounced
+in the sentence.
+
+There are many parameters for all tasks, really check out the docs to find them.
+For instance this pipeline also handles `chunk_length_s` which is helpful to work on 
+really long audio files which the models cannot handle on their own. This is helpful when
+subtitling entire movies or hour-long videos.
+
+
+Cannot you find a parameter which would really help you out ?
+[Request it](https://github.com/huggingface/transformers/issues/new?assignees=&labels=feature&template=feature-request.yml)
+
+
+## Using pipelines on a dataset
+
+So you have a large dataset and want to run inference on it to get something like a
+metric for it, perfect !
+
+The easiest and more generic way we recomment is using an iterator.
 
 ```py
->>> from transformers import pipeline
+def data():
+    for i in range(1000):
+        yield f"My example {i}"
 
->>> audio_classifier = pipeline(
-...     task="audio-classification", model="ehcalabres/wav2vec2-lg-xlsr-en-speech-emotion-recognition"
-... )
+
+pipe = pipe(model="gpt2", device=0)
+generated_characters = 0
+for out in pipe(data()):
+    generated_characters += len(out["generated_text"])
 ```
 
-Pass the audio file to the [`pipeline`]:
+The iterator `data()` will yield each result. The pipeline will automatically
+recognize that the input is iterable, and will start fetching the data while
+it continues to process it on the GPU. This uses [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) under the hood.
+This is important so you don't have to allocate memory for the whole dataset
+and you can feed the GPU as fast as possible.
+
+Since batching could speed things up, using `batch_size` here and tuning it could be important.
+
+Turns out we also have a fantasic library for [datasets](https://github.com/huggingface/datasets/).
+which makes this even simpler.
 
 ```py
->>> preds = audio_classifier(audio_file)
->>> preds = [{"score": round(pred["score"], 4), "label": pred["label"]} for pred in preds]
->>> preds
-[{'score': 0.1315, 'label': 'calm'}, {'score': 0.1307, 'label': 'neutral'}, {'score': 0.1274, 'label': 'sad'}, {'score': 0.1261, 'label': 'fearful'}, {'score': 0.1242, 'label': 'happy'}]
+# KeyDataset is a util that will just output the item we're interested in.
+from transformers.pipelines.pt_utils import KeyDataset
+
+pipe = pipeline(model="hf-internal-testing/tiny-random-wav2vec2", device=0)
+dataset = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation[:10]")
+
+for out in pipe(KeyDataset(dataset["audio"])):
+    print(out)
 ```
 
-## Vision pipeline
+
+## Using pipelines for a webserver
+
+<Tip>
+Creating an inference engine is a complex topic which deserves it's own
+page.
+</Tip>
+
+[Link](./pipeline_webserver)
+
+### Vision pipeline
```

Comment:
This section and the ones below can be ##-level headers since they don't really fit under the topic above.

```suggestion
## Vision pipeline
```

Reply:
Oops I think you forgot to change the section level for the Text and Multimodal sections!